Integration tests fail on macOS

The parallel dataloader tests still fail on macOS:
```console
$ pytest -m slow
...
self = <tests.samplers.test_single.TestGridGeoSampler object at 0x13524a160>, sampler = <torchgeo.samplers.GridGeoSampler object at 0x13524a2e0>, num_workers = 2

    @pytest.mark.slow
    @pytest.mark.parametrize("num_workers", [0, 1, 2])
    def test_dataloader(self, sampler: GridGeoSampler, num_workers: int) -> None:
        ds = CustomGeoDataset()
        dl = DataLoader(
            ds, sampler=sampler, num_workers=num_workers, collate_fn=stack_samples
        )
>       for _ in dl:

tests/samplers/test_single.py:155: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../.spack/.spack-env/view/lib/python3.8/site-packages/torch/utils/data/dataloader.py:359: in __iter__
    return self._get_iterator()
../.spack/.spack-env/view/lib/python3.8/site-packages/torch/utils/data/dataloader.py:305: in _get_iterator
    return _MultiProcessingDataLoaderIter(self)
../.spack/.spack-env/view/lib/python3.8/site-packages/torch/utils/data/dataloader.py:918: in __init__
    w.start()
../.spack/.spack-env/._view/upu5ollpmvlcwd5wntc5ajlk2y3qkjpg/lib/python3.8/multiprocessing/process.py:121: in start
    self._popen = self._Popen(self)
../.spack/.spack-env/._view/upu5ollpmvlcwd5wntc5ajlk2y3qkjpg/lib/python3.8/multiprocessing/context.py:224: in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
../.spack/.spack-env/._view/upu5ollpmvlcwd5wntc5ajlk2y3qkjpg/lib/python3.8/multiprocessing/context.py:284: in _Popen
    return Popen(process_obj)
../.spack/.spack-env/._view/upu5ollpmvlcwd5wntc5ajlk2y3qkjpg/lib/python3.8/multiprocessing/popen_spawn_posix.py:32: in __init__
    super().__init__(process_obj)
../.spack/.spack-env/._view/upu5ollpmvlcwd5wntc5ajlk2y3qkjpg/lib/python3.8/multiprocessing/popen_fork.py:19: in __init__
    self._launch(process_obj)
../.spack/.spack-env/._view/upu5ollpmvlcwd5wntc5ajlk2y3qkjpg/lib/python3.8/multiprocessing/popen_spawn_posix.py:47: in _launch
    reduction.dump(process_obj, fp)
../.spack/.spack-env/._view/upu5ollpmvlcwd5wntc5ajlk2y3qkjpg/lib/python3.8/multiprocessing/reduction.py:60: in dump
    ForkingPickler(file, protocol).dump(obj)
torchgeo/datasets/geo.py:193: in __getstate__
    objects = self.index.intersection(self.index.bounds, objects=True)
../.spack/.spack-env/view/lib/python3.8/site-packages/rtree/index.py:680: in intersection
    return self._intersection_obj(coordinates, objects)
../.spack/.spack-env/view/lib/python3.8/site-packages/rtree/index.py:722: in _intersection_obj
    p_mins, p_maxs = self.get_coordinate_pointers(coordinates)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = rtree.index.Index(bounds=[1.7976931348623157e+308, -1.7976931348623157e+308, 1.7976931348623157e+308, -1.7976931348623157e+308, 1.7976931348623157e+308, -1.7976931348623157e+308], size=0)
coordinates = [1.7976931348623157e+308, 1.7976931348623157e+308, 1.7976931348623157e+308, -1.7976931348623157e+308, -1.7976931348623157e+308, -1.7976931348623157e+308]

    def get_coordinate_pointers(self, coordinates):
    
        try:
            iter(coordinates)
        except TypeError:
            raise TypeError('Bounds must be a sequence')
        dimension = self.properties.dimension
    
        mins = ctypes.c_double * dimension
        maxs = ctypes.c_double * dimension
    
        if not self.interleaved:
            coordinates = Index.interleave(coordinates)
    
        # it's a point make it into a bbox. [x, y] => [x, y, x, y]
        if len(coordinates) == dimension:
            coordinates += coordinates
    
        if len(coordinates) != dimension * 2:
            raise core.RTreeError(
                "Coordinates must be in the form "
                "(minx, miny, maxx, maxy) or (x, y) for 2D indexes")
    
        # so here all coords are in the form:
        # [xmin, ymin, zmin, xmax, ymax, zmax]
        for i in range(dimension):
            if not coordinates[i] <= coordinates[i + dimension]:
>               raise core.RTreeError(
                    "Coordinates must not have minimums more than maximums")
E               rtree.exceptions.RTreeError: Coordinates must not have minimums more than maximums

../.spack/.spack-env/view/lib/python3.8/site-packages/rtree/index.py:357: RTreeError
```
This bug was supposed to be fixed by #304, but this appears to be a new manifestation of #184. Also see https://github.com/Toblerity/rtree/issues/87 for details.

We should:

1. Fix this bug
2. Test macOS and Windows on the release branch

--- a/tests/samplers/test_batch.py
+++ b/tests/samplers/test_batch.py
@@ -39,7 +39,9 @@ class CustomGeoDataset(GeoDataset):
 class TestBatchGeoSampler:
     @pytest.fixture(scope="class")
     def dataset(self) -> CustomGeoDataset:
-        return CustomGeoDataset()
+        ds = CustomGeoDataset()
+        ds.index.insert(0, (0, 100, 200, 300, 400, 500))
+        return ds
 
     @pytest.fixture(scope="function")
     def sampler(self) -> CustomBatchGeoSampler:

--- a/tests/samplers/test_single.py
+++ b/tests/samplers/test_single.py
@@ -39,7 +39,9 @@ class CustomGeoDataset(GeoDataset):
 class TestGeoSampler:
     @pytest.fixture(scope="class")
     def dataset(self) -> CustomGeoDataset:
-        return CustomGeoDataset()
+        ds = CustomGeoDataset()
+        ds.index.insert(0, (0, 100, 200, 300, 400, 500))
+        return ds
 
     @pytest.fixture(scope="function")
     def sampler(self) -> CustomGeoSampler:

